game/campaignloader/campaignairwingconfig.py

```diff
@@ -71,9 +71,25 @@ def from_campaign_data(
         cls, data: dict[Union[str, int], Any], theater: ConflictTheater
     ) -> CampaignAirWingConfig:
         by_location: dict[ControlPoint, list[SquadronConfig]] = defaultdict(list)
+        carriers = theater.find_carriers()
+        lhas = theater.find_lhas()
         for base_id, squadron_configs in data.items():
             if isinstance(base_id, int):
                 base = theater.find_control_point_by_airport_id(base_id)
+            elif "CVN" in base_id:
+                if carriers:
+                    base = carriers.pop(0)
```

Comment:
Is it possible that we assign a squadron to an enemy carrier here? 🤔

Reply:
Yes, this is now fixed, the code is a little messy but hopefully not too messy for your taste haha